saleor/graphql/core/utils/error_codes.py

```diff
@@ -0,0 +1,225 @@
+    REQUIRED = "required"
+    SHIPPING_ADDRESS_NOT_SET = "shipping_address_not_set"
+    SHIPPING_METHOD_NOT_APPLICABLE = "shipping_method_not_applicable"
+    SHIPPING_METHOD_NOT_SET = "shipping_method_not_set"
+    SHIPPING_NOT_REQUIRED = "shipping_not_required"
+    TAX_ERROR = "tax_error"
+    UNIQUE = "unique"
+    VOUCHER_NOT_APPLICABLE = "voucher_not_applicable"
+    ZERO_QUANTITY = "zero_quantity"
+
+
+class MenuErrorCode(Enum):
+    CANNOT_ASSIGN_NODE = "cannot_assign_node"
+    GRAPHQL_ERROR = "graphql_error"
+    INVALID = "invalid"
+    INVALID_MENU_ITEM = "invalid_menu_item"
+    NO_MENU_ITEM_PROVIDED = "no_item_provided"
+    NULL = "null"
+    OBJECT_DOES_NOT_EXIST = "object_does_not_exist"
+    REQUIRED = "required"
+    TOO_MANY_MENU_ITEMS = "too_many_items"
+    UNIQUE = "unique"
+
+
+class OrderErrorCode(Enum):
+    CANNOT_CANCEL_FULFILLMENT = "cannot_cancel_fulfillment"
+    CANNOT_CANCEL_ORDER = "cannot_cancel_order"
+    CANNOT_DELETE = "cannot_delete"
+    CANNOT_REFUND = "cannot_refund"
+    CAPTURE_INACTIVE_PAYMENT = "capture_inactive_payment"
+    EDIT_NON_DRAFT_ORDER = "edit_non_draft_order"
+    FULFILL_ORDER_LINE = "fulfill_order_line"
+    GRAPHQL_ERROR = "graphql_error"
+    INVALID = "invalid"
+    NULL = "null"
+    OBJECT_DOES_NOT_EXIST = "object_does_not_exist"
+    ORDER_INVALID_SHIPPING_METHOD = "order_invalid_shipping_method"
+    ORDER_NO_PRODUCTS = "order_without_products"
+    ORDER_NO_SHIPPING_ADDRESS = "order_no_shipping_address"
+    PAYMENT_ERROR = "payment_error"
+    PAYMENT_MISSING = "payment_missing"
+    REQUIRED = "required"
+    SHIPPING_METHOD_REQUIRED = "shipping_method_required"
+    UNIQUE = "unique"
+    VOID_INACTIVE_PAYMENT = "void_inactive_payment"
+    ZERO_QUANTITY = "zero_quantity"
+
+
+class ProductErrorCode(Enum):
+    ATTRIBUTE_ALREADY_ASSIGNED = "attribute_already_assigned"
+    ATTRIBUTE_BAD_VALUE = "attribute_bad_value"
+    ATTRIBUTE_CANNOT_BE_ASSIGNED = "attribute_cannot_be_assigned"
+    ATTRIBUTE_DISABLED_VARIANTS = "attribute_disabled_variants"
+    ATTRIBUTE_NON_ASSIGNABLE = "attribute_non_assignable"
+    ATTRIBUTE_VALUES_NOT_UNIQUE = "attribute_values_not_unique"
+    GRAPHQL_ERROR = "graphql_error"
+    INCORRECT_VALUE = "incorrect_value"
+    INVALID = "invalid"
+    MISSING_CONFIGURATION_FIELDS = "missing_configuration_fields"
+    MISSING_VALUE = "missing_value"
+    NOT_PRODUCTS_IMAGE = "not_products_image"
+    NULL = "null"
+    OBJECT_ALREADY_EXISTS = "object_already_exists"
+    OBJECT_DOES_NOT_EXIST = "object_does_not_exist"
+    REQUIRED = "required"
+    UNIQUE = "unique"
+    VALUE_ERROR = "value_error"
+    VARIANT_NO_DIGITAL_CONTENT = "variant_no_digital_content"
+
+
+class ShopErrorCode(Enum):
+    CANNOT_FETCH_TAX_RATES = "cannot_fetch_tax_rates"
+    GRAPHQL_ERROR = "graphql_error"
+    INVALID = "invalid"
+    NULL = "null"
+    OBJECT_ALREADY_EXISTS = "object_already_exists"
+    OBJECT_DOES_NOT_EXIST = "object_does_not_exist"
```

Comment:
Those two look strangely generic.

Reply:
Is it bad?

Reply:
When would we return these? Not just for this enum, for all enums. There's usually a better way to indicate what went wrong like returning `CUSTOMER_ALREADY_EXISTS` or returning `ALREADY_EXISTS` accompanied by a proper field name.

Reply:
I think all necessary information is given with that error code by attaching it to the correct field. However, I've replaced `OBJECT_ALREADY_EXISTS` with more specific error codes, as you suggested.

Refactoring of `OBJECT_DOES_NOT_EXIST ` is more complicated. I think e.g. `get_node_or_error` should be rewritten and should create field-specific codes - or we have to remain some generic error code for that. I think it could be done via separated PR.

Reply:
Yeah, I guess `get_node_or_error` should start accepting error codes.

Reply:
But is the current implementation acceptable or is this something that we definitely want to change before merging?